### ISSUE TITLE
Remove 'Edit' button from message options for chat with restricted model/app

### ISSUE THREAD

**Issue by YauheniyaH** [epam/ai-dial-chat#2784]
### EPAM AI DIAL chat version

0.22.0

### How to reproduce

Preconditions:
create custom app
create chat with history based on custom app

Steps:
1. delete custom app
2. find chat from precondition 
3. click Edit icon for any message
4. update message and click "Save & Submit" 
5. open the same chat in mobile view and tap on message

### Actual result

4. chat is displayed as generating response , but it stuck in this state.
in console there is bad request error for POST /api/chat
5. context menu is not available for mobile view when chat has restricted model/app

### Expected result

4. Edit option should not be available for messages if chat has restricted model/app
5. add context menu with options: Delete, Set message template for mobile view for chat with restricted model/app

### Confidential information

- [X] I confirm that do not share any confidential information

**Comment by VolhaBazhkova** [comment]
Return the ticket to developer as the second part concerning the mobile is not fixed.
Please see the 5th step and expected: add context menu with options: Delete, Set message template for mobile view for chat with restricted model/app
@Gimir

**Comment by VolhaBazhkova** [comment]
Agreed with developer not to disable the button, but fully remove in this case. It will be better design for model and consistent with copy button (which is not shown in case it's unuseful)

**Comment by VolhaBazhkova** [comment]
Verified on staging env successfully
